### PR TITLE
[Neo DevPack Compiler] fix type issue and test issue

### PR DESCRIPTION
**Pull request by Jim8y**
We can not properly compile nullable types, they are compiled to `List` (Array) types and is not correct, thus having this pr that compiles nullable value to `Any`.

And also enables the test engine to update multiple artifacts instead of fail on every artifact update.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/Helper.cs [hunk 1]
```diff
@@ -63,6 +63,7 @@ public static bool IsInternalCoreMethod(this IMethodSymbol method)
 
         public static ContractParameterType GetContractParameterType(this ITypeSymbol type)
         {
+            if (type.ToString()!.EndsWith('?')) return ContractParameterType.Any;
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/Helper.cs:66]: All the types inside VM are nullable, we can lost information here

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/Helper.cs:66]: removed this change, will demonstrate in another pr. So this pr will only be allowing the test engine to update multiple artifacts and run them in parallel.

File: tests/Neo.Compiler.CSharp.UnitTests/TestCleanup.cs [hunk 2]
```diff
@@ -58,29 +63,53 @@ internal static CompilationContext[] EnsureArtifactsUpToDateInternal()
...
+                            }
+                        }
+                        else
+                        {
+                            updatedArtifactNames.TryAdd(res!);
+                        }
+                    });
+                }).ToArray();
+
+            Task.WhenAll(tasks).Wait();
```
Comment (cschuchardt88) [inline tests/Neo.Compiler.CSharp.UnitTests/TestCleanup.cs:99]:
```suggestion
            Task.WhenAll(tasks).GetAwaiter().GetResult();
```

**Review by cschuchardt88 — DISMISSED** [review]

**Review by cschuchardt88 — APPROVED** [review]

**Comment by Jim8y** [comment]
@Hecate2 now you can run the tests twice and update all artifacts.

**Review by shargon — COMMENTED** [review]
LgTM

**Pull request merged**